packages/governance/test/swingsetTests/contractGovernor/test-governor.js

```diff
@@ -155,10 +155,10 @@ const expectedBrokenUpdateLog = [
   '&& running a task scheduled for 2. &&',
   'vote outcome: {"changes":{"Electorate":{"brand":"[Alleged: Zoe Invitation brand]","value":[{"description":"questionPoser","handle":"[Alleged: InvitationHandle]","installation":"[Alleged: Installation]","instance":"[Alleged: InstanceHandle]"}]},"MalleableNumber":"[42n]"}}',
   'Validation complete: true',
-  'vote rejected outcome: Error: (an object) was not a live payment for brand (an object). It could be a used-up payment, a payment for another brand, or it might not be a payment at all.',
-  'update failed: Error: (an object) was not a live payment for brand (an object). It could be a used-up payment, a payment for another brand, or it might not be a payment at all.',
-  'Electorate,MalleableNumber changed in a vote.',
+  'vote rejected outcome: Error: A Zoe invitation is required, not (an object)',
```

Comment:
yeah why did that test change that way?

Reply:
I am not landing this PR until I figure it out. This is just weird.

Reply:
in  paramManager.js, we see
```
    const prepareToSetInvitation = async invite => {
      const [preparedAmount] = await Promise.all([
        E(E(zoe).getInvitationIssuer()).getAmountOf(invite),
        assertInvitation(invite),
      ]);

      return [invite, preparedAmount];
    };
```
`E(E(zoe).getInvitationIssuer()).getAmountOf(invite)` and `assertInvitation(invite)` do slightly different checks. `getAmountOf()` calls `assertLivePayment`, which rejects with "was not a live payment for ...".

`assertInvitation` calls `E(zoe).getInvitationDetails`, which also calls `getAmountOf(invite)`, but catches errors and returns "A Zoe invitation is required, ...".

So changing the order or precedence of `getAmountOf` and `assertInvitation` in the two calls in the `Promise.all` could have that effect.

Reply:
Ah so this is a case that `Promise.all` shorts circuits on errors, and we effectively ended up with a race. That's one reason I hate the `Promise.all` error short-circuiting and wrote a [`PromiseAllOrErrors`](https://github.com/Agoric/testnet-load-generator/blob/b9ffa47ddfe8a509b250532f449c180b020a1c22/runner/lib/helpers/async.js#L34-L54) for that (main reason is so that I don't unexpectedly swallow errors).

Reply:
I have added a comment in the test as to the reason of the change, and that it may change again in the future.

In general we'll need to find a way to make these "log" based tests more flexible, as there will definitely be more order changes in the future.